### ISSUE TITLE
Please update gcloud

### ISSUE THREAD

**Issue by dantman** [pkgcloud/pkgcloud#452]
Please update to `gcloud@0.16.0`.

Users of `pkgcloud` (whether they use `gcloud` or not) are getting npm warnings of a deprecated version of `jws` because `pkgcloud` depends on `gcloud@0.10.0` which depends on an old `gapitoken@0.1.4` which depends on `jws@0.0.2`. The new version of gcloud doesn't even use `gapitoken`.

**Comment by indexzero** [comment]
Can you rephrase your issue in the form of a pull-request?

**Comment by dantman** [comment]
Sorry, but I cannot test the gcloud portions of the code.

**Comment by indexzero** [comment]
Not sure I follow. I get PRs all the time where authors don't bother running or writing any tests :smiling_imp:

**Comment by stephenplusplus** [comment]
gcloud is currently at version 0.30. I'll work on a PR soon to update it from 0.10.

**Comment by BobDickinson** [comment]
I was contemplating doing this myself, but I stopped short when I noticed that gcloud (starting around 0.15) started including a component (sse4_crc32 ) with native code.  We had a lot of problems when our product included native code (fibers).  It was particularly problematic on Windows and in cloud environments (Azure and some others).  It was such a bag of fail that we actually ended up doing a non-trivial port away from fibers mainly to get away from the native code and build issues.  

As just one example, we had one trade mag writer do a deep review of our product at launch, and he happened to be a Windows guy.  Our product install failed for him, because he did not have a proper build environment (which is the norm for Windows Node users).  Kind of hard to overcome the fact that the product doesn't install/run.

Not sure the gcloud team is overly concerned about the idea of having native code (or maybe they just really love fast CRCs).  Here is one of their sse4_crc32 issues:

https://github.com/GoogleCloudPlatform/gcloud-node/issues/1197

I get the impression that the install should still work if the build fails.  I'm going to try to test installing the latest gcloud on a vanilla Windows image (no build tools) and on Azure within the next day or two.  I guess if it fails cleanly, then I'd be OK with it.

**Comment by stephenplusplus** [comment]
Yeah, we switched from relying on `sse4_crc32` directly, instead using [`fast-crc32c`](https://github.com/ashi009/node-fast-crc32c). It marks `sse4_crc32` as an optional dependency, so if it fails, it uses its own software implementation.

**Comment by BobDickinson** [comment]
Well, I guess an optional dependency is better than a hard dependency on native code.  Optional dependencies are still not without their problems.  Here, for example, is a 77 participant issue that resulted largely from people not understanding the warning message the npm gives on a failed optional dependency:

https://github.com/npm/npm/issues/9204

An issue which affects my product is that optional dependencies are a pain when you use npm shrinkwrap (which is common when distributing products versus libs/tools).  Shrinkwrap doesn't know anything about optional dependencies:

https://github.com/npm/npm/issues/2679

So if you use shrinkwrap, there are no optional dependencies.  If you shrinkwrap without building sse4_crc32, then nobody gets it ever (which is fine with me, and what we'll have to do).

I really just don't like native code in Node packages, but we definitely need contemporary gcloud, so I guess we don't have a choice.
